smithy-aws-iam-traits/src/main/resources/META-INF/smithy/aws.iam.smithy

```diff
@@ -34,6 +36,29 @@ map defineConditionKeys {
 @trait(selector: ":test(service, resource)")
 structure disableConditionKeyInference {}
 
+/// Indicates properties of a Smithy operation as an IAM action.
+@trait(selector: "operation")
+structure iamAction {
+    /// The name of the action in AWS IAM.
+    name: String
+
+    /// A brief description of what granting the user permission to invoke an operation would entail.
+    /// This description should begin with something similar to 'Enables the user to...' or 'Grants permission to...'
+    documentation: String
+
+    /// A relative URL path that defines more information about the action within a set of IAM-related documentation.
+    relativeDocumentation: String
+
+    /// Other actions that the invoker must be authorized to perform when executing the targeted operation.
+    requiredActions: RequiredActionsList
+
+    /// The resources an IAM action can be authorized against.
+    resources: ActionResources
+
+    /// The resources that performing this IAM action will create.
+    createsResources: ResourceNameList
```

Comment:
Should a user still need to supply this for an operation bound to `create` lifecycle for a resource if it doesnt create any other resources?